src/main/java/io/fabric8/elasticsearch/plugin/discovery/kubernetes/KubernetesDiscoveryPlugin.java

```diff
@@ -53,12 +53,13 @@ public static boolean isDiscoveryAlive(Settings settings, ESLogger logger) {
       return false;
     }
 
-    if (!checkProperty(KubernetesAPIService.Fields.NAMESPACE, settings.get(KubernetesAPIService.Fields.NAMESPACE), logger) ||
-      !checkProperty(KubernetesAPIService.Fields.SERVICE_NAME, settings.get(KubernetesAPIService.Fields.SERVICE_NAME), logger)) {
+    if ( !hasNamespace(settings, logger) || !(hasServiceName(settings, logger) ^ hasPodLabel(settings, logger)) ) {
```

Comment:
That depends on what the desired behavior is. The `^` is an exclusive or, so it should prevent the definition of both service name and pod label. In this case, I think we should not allow defining both in the config file.